### PR TITLE
Action delete cluster by name

### PR DESCRIPTION
**Pull request by desistefanova**
Deletes a cluster by given specific name.
Done in https://github.com/realm/ci-actions/pull/19

### CODE REVIEW

File: mdb-realm/deleteCluster/action.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    description: The Atlas project Id.
+  apiKey:
+    required: true
+    description: The public API key.
+  privateapiKey:
+    required: true
+    description: The private API key.
+  clusterName:
+    required: false
+    description: Provide a specific name of cluster that to be deleted."
```
Comment (nirinchev) [inline mdb-realm/deleteCluster/action.yml:22]:
```suggestion
    description: Provide a specific name of cluster that to be deleted.
```

**Review by nirinchev — CHANGES_REQUESTED** [review]
Do we need this? I guess most of this functionality is covered by the cleanup action - the only difference is that it also deletes the applications for that cluster. Is there a case where you only want to delete the cluster but not the apps?

**Comment by desistefanova** [comment]
> Do we need this? I guess most of this functionality is covered by the cleanup action - the only difference is that it also deletes the applications for that cluster. Is there a case where you only want to delete the cluster but not the apps?

Yes @nirinchev, we delete the cluster and the apps at the end of the workflow. The only thing that cleanup and deleteAllClusters are missing is the parameter clusterName. They both don't support deleting a specific cluster. They will delete all the clusters for all the running CIs in this project. We are running all the PR's CIs in the same project.
BTW, What is the difference between cleanup and deleteAllClusters? It looks they are doing the same thing.

**Comment by nirinchev** [comment]
cleanup calls `getConfig`, which will get the cluster name if provided as an argument: https://github.com/realm/ci-actions/blob/0101f67e2a3cc2e1f46087f8264edc7ef0aa4aa3/mdb-realm/src/helpers.ts#L23. It looks like we should add it to the action.yml here: https://github.com/realm/ci-actions/blob/0101f67e2a3cc2e1f46087f8264edc7ef0aa4aa3/mdb-realm/cleanup/action.yml#L4 and it should just work. `deleteAllClusters` deletes all clusters, not an individual one.

**Comment by desistefanova** [comment]
> > Do we need this? I guess most of this functionality is covered by the cleanup action - the only difference is that it also deletes the applications for that cluster. Is there a case where you only want to delete the cluster but not the apps?
> 
> Yes @nirinchev, we delete the cluster and the apps at the end of the workflow. The only thing that cleanup and deleteAllClusters are missing is the parameter clusterName. They both don't support deleting a specific cluster. They will delete all the clusters for all the running CIs in this project. We are running all the PR's CIs in the same project. BTW, What is the difference between cleanup and deleteAllClusters? It looks they are doing the same thing.

@nirinchev OK, I can do that and then I will close this PR.

**Comment by desistefanova** [comment]
> Done in #19

**Comment by desistefanova** [comment]
Moved to https://github.com/realm/ci-actions/pull/19

**Pull request closed without merging**